src/olympia/devhub/forms.py

```diff
@@ -165,24 +165,27 @@ def get_tags(self, addon):
                         .values_list('tag_text', flat=True))
 
     def save(self, *args, **kwargs):
-        metadata_content_review = waffle.switch_is_active(
-            'metadata-content-review')
+        metadata_content_review = (waffle.switch_is_active(
+            'metadata-content-review') and
+            self.instance and
+            self.instance.has_listed_versions())
         existing_data = (
             fetch_existing_translations_from_addon(
                 self.instance, self.fields_to_trigger_content_review)
-            if self.instance and metadata_content_review else {})
+            if metadata_content_review else {})
         obj = super().save(*args, **kwargs)
         if not metadata_content_review:
             return obj
-        new_data = (
-            fetch_existing_translations_from_addon(
-                obj, self.fields_to_trigger_content_review)
-        )
-        if existing_data != new_data:
-            # flag for content review
-            statsd.incr('devhub.metadata_content_review_triggered')
-            AddonApprovalsCounter.reset_content_for_addon(addon=obj)
-        return obj
+        else:
+            new_data = (
+                fetch_existing_translations_from_addon(
+                    obj, self.fields_to_trigger_content_review)
+            )
+            if existing_data != new_data:
+                # flag for content review
+                statsd.incr('devhub.metadata_content_review_triggered')
+                AddonApprovalsCounter.reset_content_for_addon(addon=obj)
+            return obj
```

Comment:
you don't technically need any of this code churn now - `metadata_content_review` would be False so execution would have returned already.